src/Symfony/Bridge/Doctrine/Security/MagicLink/MagicLoginLinkTokenEntityTrait.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Security\MagicLink;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * Trait that can be used by your entity to help implement StoredMagicLinkTokenInterface.
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+trait MagicLoginLinkTokenEntityTrait
+{
+    /**
+     * @ORM\Column(type="string", length=20)
+     */
+    private $selector;
+
+    /**
+     * @ORM\Column(type="string", length=100)
+     */
+    private $hashedVerifier;
+
+    /**
+     * @ORM\Column(type="datetime_immutable")
+     */
+    private $createdAt;
+
+    /**
+     * @ORM\Column(type="datetime_immutable")
+     */
+    private $expiresAt;
+
+    /**
+     * Call this from your constructor to initialize the fields.
+     */
+    private function initialize(\DateTimeInterface $expiresAt, string $selector, string $hashedVerifier)
```

Comment:
this should accept only `DateTimeImmutable` IMO, forcing the usage of the immutable value object.

Reply:
I don't know, it feels weird to force a stronger interface, when I don't actually need any of the methods on that stronger interface.